src/test/java/io/camunda/testing/util/Utilities.java

```diff
@@ -77,13 +76,15 @@ public static DeploymentEvent deployProcesses(
   }
 
   public static ProcessInstanceEvent startProcessInstance(
-      final ZeebeClient client, final String processId) throws InterruptedException {
-    return startProcessInstance(client, processId, new HashMap<>());
+      final ZeebeEngine engine, final ZeebeClient client, final String processId) {
+    return startProcessInstance(engine, client, processId, new HashMap<>());
```

Comment:
This pull request makes considerable changes to the public interface, mainly by adding `engine` as an additional parameter to these methods. That is of course necessary in order to use the `waitForIdleState` method, but with the number of parameters increasing we could consider improving usability, moving common parameters to a constructor and no longer having static methods.

❓ Is this something you have considered? Do we have a goal design in mind for the interfaces? 

I ask this because changing such interfaces after an initial release might be difficult due to breaking changes.

Reply:
As mentioned in the comment above, this is not a public interface. Therefore I don't see a problem in the way it is currently. Eventually we will build a public interface and at that point we will also start thinking it through more thoroughly, and this code will probably be replaced.

This is not something that will be done this quarter though.

Reply:
Never mind my carefulness. 👍

See comment in above conversation.